game-backend/src/MainController.ts

```diff
@@ -1,18 +1,63 @@
 import WebSocket from 'ws';
 import hyperid from 'hyperid';
 import { GameObject } from '../../darwin-types/GameObject';
+import { UserContext } from '../../darwin-types/UserContext';
+import { ServerStore, createStore } from './ServerStore';
 
-const hyperIdInstance = hyperid();
-
+/**
+ * Main controller, which handles new connections and stores match data and other server data.
+ */
 export default class MainController {
-  static newConnection(ws: WebSocket): void {
+  private hyperIdInstance: hyperid.Instance = hyperid();
+
+  private store: ServerStore = createStore();
+
+  private isTicking = false;
+
+  private tickingInterval: NodeJS.Timeout;
+
+  newConnection(ws: WebSocket, connectionId: string): void {
+    this.store.connections.push([connectionId, ws]);
+
+    // generate a unit
+    const unitId = this.hyperIdInstance();
     const unit: GameObject = {
-      id: hyperIdInstance(),
+      id: unitId,
       position: {
         x: Math.floor(Math.random() * 21),
         y: Math.floor(Math.random() * 21),
       },
     };
-    ws.send(JSON.stringify(unit));
+
+    // add it to the match state
+    this.store.matchState.objectMap[unit.id] = unit;
+    this.store.matchState.objectIds.push(unit.id);
+
+    // add the generated unit to the user context
+    const userCtx: UserContext = { unitId };
+    this.store.userContexts.userContextMap[connectionId] = userCtx;
+    this.store.userContexts.userContextIds.push(connectionId);
+
+    if (!this.isTicking && this.store.connections.length) {
+      this.startTicking();
+    }
+  }
+
+  static newMessage(msg: WebSocket.Data, ws: WebSocket, id: string): void {
+    ws.send(`got message ${msg} from ${id}`);
+  }
```

Comment:
@sir-marc Same as for the "unnecessary" changes in `server.ts` -> Move Out of this PR?

Reply:
Jep, same here.